### ISSUE TITLE
URGENT: Code for R6 active bindings is evaluated on `$` autocompletion.

### ISSUE THREAD

**Issue by wurli** [rstudio/rstudio#14784]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2024.04.1 Build 748
    OS Version      : Windows 11
    R Version       : 4.3.1

### Steps to reproduce the problem
Run the following code:
``` R
Test <- R6::R6Class("Test",
  active  = list(active_test  = function(value) print("active")),
  private = list(private_test = function(value) print("private")),
  public  = list(public_test  = function(value) print("public"))
)

n <- Test$new()
```
Then, typing `n$` to trigger autocomplete, `[1] "active"` will be printed to the console, indicating that the code for the active binding function has been run without the user's intent. Here is a demonstration: https://github.com/rstudio/rstudio/assets/17475731/70169848-5350-4f14-8b23-a8a86c5c6b20

This bug has potential security implications as it results in R code being evaluated without the user's intent. For my team, this has caused some serious problems as code has been run during auto-completion which has affected a remote database. In a project working with a particularly complex R6 class, autocompletion results in so much code getting run (and so many messages being printed to the console) that we needed to roll back our version of RStudio to continue working.

### Describe the behavior you expected
Autocomplete should work without any code being run.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by wurli** [comment]
Unfortunately I can't install the daily on my Windows (work) machine, but can confirm the issue seems to be fixed on macOS. Thanks for the speedy response!

**Comment by wurli** [comment]
Can confirm it's also fixed the issues we've been experiencing on Windows :) Many thanks!

**Comment by wurli** [comment]
> @kevinushey - apologies as cannot test at the moment but any chance this potentially fixes #9692 as well?

@TimTaylor I gave this a quick test and looks like the issue is persisting, so probably unrelated unfortunately

**Comment by ronblum** [comment]
Original issue verified in RStudio Desktop 2024.07.0-daily+244 on Windows 11.
